java/arcs/core/host/Utils.kt

```diff
@@ -17,3 +22,31 @@ fun KClass<*>.className(): String {
         .substringBefore('<')
         .replace('$', '.')
 }
+
+/** Returns a pair mapping [ParticleIdentifier] to [ParticleConstructor] */
+inline fun <reified T : Particle> (() -> T).toRegistration(): ParticleRegistration =
+    T::class.toParticleIdentifier() to suspend { this.invoke() }
+
+/**
+ * If this Type represents a [SingletonType], [CollectionType], or [EntityType], return the
+ * [Schema] used by the underlying [Entity] that this type represents.
+ */
+fun Type.toSchema(): Schema {
+    when (this) {
+        is SingletonType<*> -> if (this.containedType is EntityType) {
```

Comment:
I'm curious what will happen if a SingletonType contains something that is not an EntityType? What should happen?

It looks like it will fall through to the IllegalArgumentException below. Do we always expect SingletonTypes to contain EntityTypes? If so, this if-statement could become a `require` instead, with a better error message.